CodeQL-Build fails on every run.

The error is:
```
Teardown error: System.InvalidOperationException: The context data has not been setup.
   at Cake.Core.CakeDataService.Get[TData]() in C:\projects\cake\src\Cake.Core\CakeDataService.cs:line 32
   at Cake.Core.CakeEngineActions.<>c__DisplayClass21_0`1.<RegisterTeardown>b__0(ITeardownContext context) in C:\projects\cake\src\Cake.Core\CakeEngineActions.cs:line 64
   at Cake.Core.DefaultExecutionStrategy.PerformTeardown(Action`1 action, ITeardownContext teardownContext) in C:\projects\cake\src\Cake.Core\DefaultExecutionStrategy.cs:line 67
   at Cake.Core.CakeEngine.PerformTeardown(IExecutionStrategy strategy, ICakeContext context, Stopwatch stopWatch, CakeReport report, Boolean exceptionWasThrown, Exception thrownException) in C:\projects\cake\src\Cake.Core\CakeEngine.cs:line 494
Error: One or more errors occurred. (The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception.)
The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception.
Error: Process completed with exit code 1.
```

The underlying problem here is that the CodeQL workflow is based on `ubuntu-latest` which currently is an alias for `ubuntu-20.04` which has known issues running libgit2sharp. The fix would be to pin the workflow to `ubuntu-18.04`

--- a/.github/workflows/codeql-analysis.yml
+++ b/.github/workflows/codeql-analysis.yml
@@ -13,11 +13,12 @@ on:
     branches: [develop]
   schedule:
     - cron: '0 15 * * 6'
+  workflow_dispatch:
 
 jobs:
   analyze:
     name: Analyze
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-18.04
 
     strategy:
       fail-fast: false
@@ -30,14 +31,15 @@ jobs:
 
     steps:
     - name: Checkout repository
-      uses: actions/checkout@v2.3.4
+      uses: actions/checkout@v2
       with:
         fetch-depth: 0
 
-    # If this run was triggered by a pull request event, then checkout
-    # the head of the pull request instead of the merge commit.
-    - run: git checkout HEAD^2
-      if: ${{ github.event_name == 'pull_request' }}
+    - name: Cache Tools
+      uses: actions/cache@v2
+      with:
+        path: tools
+        key: ${{ runner.os }}-tools-${{ hashFiles('recipe.cake') }}
 
     # Initializes the CodeQL tools for scanning.
     - name: Initialize CodeQL
@@ -49,8 +51,25 @@ jobs:
         # Prefix the list here with "+" to use these queries and those in the config file.
         # queries: ./path/to/local/query, your-org/your-repo/queries@main
 
-    - run: ./build.ps1
-      shell: pwsh
+    - name: Build project
+      uses: cake-build/cake-action@v1
+      with:
+        script-path: recipe.cake
+        target: DotNetCore-Build
+        cake-version: 0.38.5
+        cake-bootstrap: true
+      env:
+        COMPlus_DbgEnableMiniDump: 1
+        COMPlus_DbgMiniDumpType: 1
+        COMPlus_DbgMiniDumpName: BuildArtifacts/coredump.dmp
 
     - name: Perform CodeQL Analysis
       uses: github/codeql-action/analyze@v1
+
+    - name: Upload CoreDump
+      uses: actions/upload-artifact@v2
+      if: failure()
+      with:
+        if-no-files-found: warn
+        name: CoreDump
+        path: BuildArtifacts/coredump.dmp

